From f6b17994e42d13e522c755a3796b038cf2838df9 Mon Sep 17 00:00:00 2001
From: Li Wencheng <liwencheng@phytium.com.cn>
Date: Fri, 26 Apr 2024 15:10:12 +0800
Subject: [PATCH 21/67] PHYTIUM: net/phytmac: Round down rx_buf_len to 64 bytes

DMA config register needs the rx_buf_len to be aligned to
64-bytes.

Mainline: NA
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Change-Id: I7535ac7d8abaf80397b39ec1a16b1b959c9decfd

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/38380eb6dd3618217d0bc254e0d4b126b331c0c1
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_main.c b/drivers/net/ethernet/phytium/phytmac_main.c
index a2b3ed84e895..72b5a15870a0 100644
--- a/drivers/net/ethernet/phytium/phytmac_main.c
+++ b/drivers/net/ethernet/phytium/phytmac_main.c
@@ -298,9 +298,9 @@ static struct net_device_stats *phytmac_get_stats(struct net_device *dev)
 static inline int phytmac_calc_rx_buf_len(void)
 {
 #if (PAGE_SIZE < 8192)
-	return PHYTMAC_MAX_FRAME_BUILD_SKB;
+	return rounddown(PHYTMAC_MAX_FRAME_BUILD_SKB, RX_BUFFER_MULTIPLE);
 #endif
-	return PHYTMAC_RXBUFFER_2048;
+	return rounddown(PHYTMAC_RXBUFFER_2048, RX_BUFFER_MULTIPLE);
 }

 inline struct phytmac_dma_desc *phytmac_get_rx_desc(struct phytmac_queue *queue,
--
2.50.1
